imports/plugins/core/versions/server/util/convert51.js

```diff
@@ -0,0 +1,484 @@
+  }
+  return variant.inventoryQuantity || 0;
+}
+
+/**
+ *
+ * @method getProductInventoryInStockQuantity
+ * @summary Get the number of product variants still avalible to sell. This calculates based off of `inventoryQuantity`.
+ * This function can take only a top product ID and a mongo collection as params to return the product
+ * `inventoryQuantity` quantity, which is a calculation of the sum of all variant `inventoryQuantity` quantities.
+ * @param {Object} productId - A top level product variant object.
+ * @param {Object} collections - Raw mongo collections.
+ * @param {Object[]} variants - Array of product variant option objects.
+ * @return {Promise<number>} Variant quantity.
+ */
+function getProductInventoryInStockQuantity(productId, collections) {
+  const variants = getVariants(productId, collections, true);
+
+  if (variants && variants.length) {
+    return variants.reduce((sum, variant) => sum + variant.inventoryQuantity || 0, 0);
+  }
+  return 0;
+}
+
+
+/**
+ *
+ * @method updateParentVariantsInventoryAvailableToSellQuantity
+ * @summary Get the number of product variants that are currently reserved in an order.
+ * This function can take any variant object.
+ * @param {Object} item - A product item object, either from the cart or the products catalog
+ * @param {Object} collections - Raw mongo collections.
+ * @return {Promise<number>} Reserved variant quantity.
+ */
+function updateParentVariantsInventoryAvailableToSellQuantity(item, collections) {
+  // Since either a cart item or a product catalog item can be provided, we need to determine
+  // the parent based on different data
+  // If this is a cart item, `productId` and `variantId` are fields on the object
+  // If this is a product object, _id is the equivalent of `variantId`, and `ancestors[0]` is the productId
+  let updateProductId;
+  let updateVariantId;
+  if (item.variantId && item.productId) {
+    updateProductId = item.productId;
+    updateVariantId = item.variantId;
+  } else {
+    updateProductId = item.ancestors[0]; // eslint-disable-line
+    updateVariantId = item._id;
+  }
+
+  // Check to see if this item is the top level variant, or an option
+  const topLevelVariant = getTopLevelVariant(updateVariantId, collections);
+  // If item is an option, update the quantity on its parent variant too
+  if (topLevelVariant._id !== updateVariantId) {
+    const variantInventoryAvailableToSellQuantity = getVariantInventoryAvailableToSellQuantity(topLevelVariant, collections);
+    const variantInventoryInStockQuantity = getVariantInventoryInStockQuantity(topLevelVariant, collections);
+
+    collections.Products.update(
+      {
+        _id: topLevelVariant._id
+      },
+      {
+        $set: {
+          inventoryAvailableToSell: variantInventoryAvailableToSellQuantity,
+          inventoryQuantity: variantInventoryInStockQuantity
+        }
+      },
+      {
+        publish: true,
+        selector: {
+          type: "variant"
+        }
```

Comment:
Rather than `selector` option, all `update` calls in migrations should pass `bypassCollection2: true` option. Otherwise when the schema continues to change in the future, these updates will eventually throw errors.